hack/update-codegen.sh

```diff
@@ -20,5 +20,48 @@ set -o pipefail
 
 source $(dirname $0)/../scripts/library.sh
 
+CONFIG_GENERATOR_DIR="${REPO_ROOT_DIR}/tools/config-generator"
```

Comment:
Would it be more clear if we move the code to a separate script called `./hack/update-configs.sh`, and run it here like how we run `./hack/update-deps.sh`?

Reply:
Then why not just keep it in `make` and call it there?

I think it makes sense that `update-codegen.sh` calls it, but I don't care as much where the code lives.

Reply:
Then developers can run `./hack/update-configs.sh` if they ONLY want to update the config files? Otherwise they will have to run `./hack/update-codegen.sh` and it takes a much longer time.

Reply:
agree with Chi, I think it makes sense moving to a separate file

Reply:
done

Reply:
It was already in a separate file, just leave it in the Makefile then...

Reply:
it's a single invocation vs. multiple invocations, I can make the scripts do 2 `make` call just to make the user experience slightly better